src/scanners/binary.js

```diff
@@ -0,0 +1,34 @@
+import BaseScanner from 'scanners/base';
+import * as messages from 'messages';
+import * as constants from 'const';
+
+
+export default class BinaryScanner extends BaseScanner {
+  static get fileStreamType() {
+    return 'chunk';
+  }
+
+  check(buffer, values) {
+    for (let v in values) {
+      if (values[v] !== buffer[v]) {
+        return;
+      }
+    }
+    this.linterMessages.push(
+      Object.assign({}, messages.FLAGGED_FILE_TYPE, {
+        type: constants.VALIDATION_NOTICE,
+        file: this.filename,
+      })
+    );
+  }
+
+  scan() {
+    return new Promise((resolve) => {
```

Comment:
You can instead return `return Promise.resolve(this.linterMessages);` at the bottom and avoid returning the function. We've been trying to do that in this repo.